crates/bevy_ecs/src/query/access.rs

```diff
@@ -41,6 +44,28 @@ impl<'a, T: SparseSetIndex + fmt::Debug> fmt::Debug for FormattedBitSet<'a, T> {
     }
 }
 
+struct FormattedExpandedOrWithAccesses<'a, T: SparseSetIndex> {
+    with: &'a ExpandedOrWithAccesses,
+    _marker: PhantomData<T>,
+}
+
+impl<'a, T: SparseSetIndex> FormattedExpandedOrWithAccesses<'a, T> {
+    fn new(with: &'a ExpandedOrWithAccesses) -> Self {
+        Self {
+            with,
+            _marker: PhantomData,
+        }
+    }
+}
+
+impl<'a, T: SparseSetIndex + fmt::Debug> fmt::Debug for FormattedExpandedOrWithAccesses<'a, T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_list()
+            .entries(self.with.arr.iter().map(FormattedBitSet::<T>::new))
+            .finish()
+    }
+}
```

Comment:
Why create a Proxy type here, instead of implementing Debug for ExpandedOrWithAccesses?
The `FormattedBitSet` is used because the `FixedBitSet` it wraps is a foreign type with a separate debug impl.

Reply:
Yeah, I copy-pasted the approach for `FixedBitSet ` without giving it much thought.. You're right, it's not needed. I had to introduce the index marker type for the debug implementation though.